src/MultivariateStats.jl

```diff
@@ -352,4 +355,165 @@ MLJBase.is_pure_julia(::Type{<:ICA}) = true
+        model.regcoef = 1e-4
+    end
+    return warning
+end
+
+
+function LDA(; method=:gevd, shrinkage=:None, out_dim=1, regcoef=1e-6)
+    model = LDA(method, shrinkage, out_dim, regcoef)
+    message = MLJBase.clean!(model)
+    isempty(message) || @warn message
+    return model
+end
+
+function MLJBase.fit(model::LDA, verbosity::Int, X, y)
+ 
+    class_list = MLJBase.classes(y[1]) #class list containing unique entries in y
+    
+    # NOTE: copy/transpose
+    Xmatrix ::Array{Float64,2} = MLJBase.matrix(X,transpose=true)
+    training_sample_size ::Int64 = size(Xmatrix,2)
+
+    #convert the target array of categoricals (y) to array of integers (yarray)
+    #and get the number of classes from yarray
+    yarray = MLJBase.int(y)
+    no_of_classes = maximum(yarray)
+
+    #check to make sure output dimension isn't too large or negative
+    0 < model.out_dim <= min(no_of_classes-1, training_sample_size) ||
+     throw(ArgumentError("out_dim must lie within 0 and min(no_of_features,training_sample_size)"))
+
+    if model.method ∈ [:gevd, :whiten]
+        model.shrinkage == :None ?
+            covestimator ::Union{MS.SimpleCovariance,LinearShrinkage}=MS.SimpleCovariance() :
+            covestimator = LinearShrinkage(target=DiagonalCommonVariance(),shrinkage=model.shrinkage)
+
+        core_fitresult=MS.fit(MS.MulticlassLDA, Int(no_of_classes), Xmatrix, Int.(yarray);
+                            method=model.method,
+                            outdim=model.out_dim,
+                            regcoef=model.regcoef,
+                            covestimator_within=covestimator,
+                            covestimator_between=covestimator)
+
+    elseif model.method == :subspace_lda
+        model.shrinkage == :None || println("Warning shrinkage does not apply to subspace_lda method")
+        core_fitresult = MS.fit(MS.SubspaceLDA, Xmatrix, Int.(yarray), no_of_classes; normalize=true)
+    
+    else
+        throw(ArgumentError("Invalid model method $(model.method)"))
+     end
+    
+    πk = proportions(yarray) #estimate prior probabilities of each class in target vector y
+    cache = nothing
+    
+    report = NamedTuple{}()
+    fitresult = (class_list, core_fitresult, πk, training_sample_size)
+    return fitresult, cache, report
+end
+
+function MLJBase.fitted_params(::LDA, (class_list, core_fitresult, πk, training_sample_size))
+    ## Note The projection matrix that projects data unto
+    ## a lower dimensional subspace and whitens the lower dimensional data .
+    return (class_means = MS.classmeans(core_fitresult),
+            projection_matrix = MS.projection(core_fitresult),
+            prior_probabilities = πk)
+end
+
+function MLJBase.predict(model::LDA , (class_list , core_fitresult , πk , training_sample_size), Xnew)
+    
+    ##Transpose the Xnew matrix and reduce its dimension
+    Xmatrix = MS.transform(core_fitresult ,MLJBase.matrix(Xnew,transpose=true)) :: Array{Float64,2}
```

Comment:
Again I'd rather we did not have such explicit typing in the code apart for very specific cases such as loops where you may otherwise risk type instability, but here it's just noise (same ocmment for rest of code)